src/components/generic/messageComponents/MessageAttachment.js

```diff
@@ -1,18 +1,100 @@
-import { motion as m } from "framer-motion";
+import { Blurhash } from "react-blurhash";
+import { Oval } from "react-loader-spinner";
+import { useLocation, useNavigate } from "react-router-dom";
+import { useMemo, useRef, useState } from "react";
+
+export default function MessageAttachment({
+  id,
+  url,
+  name,
+  localUrl,
+  blurHash,
+}) {
+  const { hash } = useLocation();
+  const navigate = useNavigate();
+
+  const [loaded, setLoaded] = useState(false);
+  const videoRef = useRef(null);
+
+  const videoView = useMemo(() => {
+    const video = (
+      <video
+        ref={videoRef}
+        controls
+        src={url + "#t=0.1"}
+        poster={name}
+        onClick={(event) => {
+          event.preventDefault();
+          videoRef.current.pause();
+        }}
+      />
+    );
+
+    //get first clip from video for bg
+
+    return video;
+  }, [url]);
+
+  const preloaderView = useMemo(() => {
```

Comment:
the same question here. Just wondering...

Reply:
agree, we could extract it into separate component for better code readability